packages/editor/src/components/post-url/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { safeDecodeURIComponent, cleanForSlug } from '@wordpress/url';
+import { useState } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { TextControl, ExternalLink } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+export default function PostURL( { onClose } ) {
+	const { isEditable, postSlug, postLink, permalinkPrefix, permalinkSuffix } =
+		useSelect( ( select ) => {
+			const permalinkParts = select( editorStore ).getPermalinkParts();
```

Comment:
Unrelated, but I think `getPermalinkParts()` should return an empty object when there're no parts, so we can safely destruct value.

Reply:
Sorry, ignore this. It would be a breaking change.

Reply:
OK, ignoring! 😀